app/src/main/kotlin/nyc/jsjrobotics/palmrgb/fragments/createMessage/CreateMessageModel.kt

```diff
@@ -1,27 +1,37 @@
 package nyc.jsjrobotics.palmrgb.fragments.createMessage
 
 import io.reactivex.Observable
-import io.reactivex.disposables.Disposable
 import io.reactivex.subjects.PublishSubject
 import nyc.jsjrobotics.palmrgb.Application
+import nyc.jsjrobotics.palmrgb.dataStructures.Message
 import nyc.jsjrobotics.palmrgb.dataStructures.Palette
+import nyc.jsjrobotics.palmrgb.firebase.SimpleAuth
+import nyc.jsjrobotics.palmrgb.firebase.datamanagers.MessageDataManager
 import nyc.jsjrobotics.palmrgb.globalState.DeviceConstants
 import nyc.jsjrobotics.palmrgb.globalState.SavedPaletteModel
 import nyc.jsjrobotics.palmrgb.service.PalmRgbBackground
-import nyc.jsjrobotics.palmrgb.service.remoteInterface.HackdayLightsInterface
+import java.util.*
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class CreateMessageModel @Inject constructor(private val application: Application,
-                                             private val savedPalettesModel: SavedPaletteModel){
+                                             private val savedPalettesModel: SavedPaletteModel,
+                                             private val simpleAuth: SimpleAuth,
+                                             private val messageDataManager: MessageDataManager) {
     var selectedPalette: Palette = savedPalettesModel.getStandardPalette()
-    var displayedColors : MutableList<Int> = initialValues() ; private set
-    val displayChanged : PublishSubject<Boolean> = PublishSubject.create()
-    val onDisplayChanged : Observable<Boolean> = displayChanged
+    var displayedColors: MutableList<Int> = initialValues(); private set
+    val displayChanged: PublishSubject<Boolean> = PublishSubject.create()
+    val onDisplayChanged: Observable<Boolean> = displayChanged
 
-    fun diodeRange() : IntRange = DeviceConstants.largeArrayRange
+    fun diodeRange(): IntRange = DeviceConstants.largeArrayRange
 
+    fun uploadMessageToFirebase(messageTitle: String) {
+        messageTitle.isNotEmpty().let {
```

Comment:
What happens if the message title is empty? Seems to me that if you call a function, it should not just drop the result. It should report an error if it is not going to complete the action. Or instead, don't do any checks here and just perform the action.